src/numbers.lisp

```diff
@@ -152,3 +157,244 @@
 	 (lcm-2 (first integers) (second integers)))
 	(t
 	 (apply #'lcm (lcm (first integers) (second integers)) (nthcdr 2 integers)))))
+
+;;; @VKM path 30.11
+
+;;; round number/division to nearest integer
+;;; second value is remainder
+(defun round (number &optional (divisor 1))
+  (multiple-value-bind (integer remainder) (truncate number divisor)
+    (if (zerop remainder)
+        (values integer remainder)
+        (let ((thresh (/ (abs divisor) 2)))
+          (cond ((or (> remainder thresh)
+                     (and (= remainder thresh) (oddp integer)))
+                 (if (minusp divisor)
+                     (values (- integer 1) (+ remainder divisor))
+                     (values (+ integer 1) (- remainder divisor))))
+                ((let ((negative (- thresh)))
+                   (or (< remainder negative)
+                       (and (= remainder negative) (oddp integer))))
+                 (if (minusp divisor)
+                     (values (+ integer 1) (- remainder divisor))
+                     (values (- integer 1) (+ remainder divisor))))
+                (t (values integer remainder)))))))
+
+(defconstant most-integer-length (expt 2 30))
+
+;;; ash
+(defun ash (x y)
+  (let* ((minus-p (minusp y))
+        (y (if minus-p (- y) y)))
+    (if minus-p
+        (%ash-right x y)
+        (%ash-left x  y))))
+
+;;; lognot
+(defun lognot (x) (%lognot x))
+
+;;; logxor
+(defun logxor (&rest others)
+  (if (null others) (return-from logxor 0)
+      (let ((integer (car others)))
```

Comment:
There are a bit of redundancy here on those functions for the iterating. 

Not a big deal but,  perhaps we could define a `macrolet` to abstract it away?

Reply:
Yes. 
This is ugly code, I agree. It is correct to use a macrolet here.
In the next release;)